src/hotspot/share/runtime/globals.hpp

```diff
@@ -1587,7 +1587,7 @@ const int ObjectAlignmentInBytes = 8;
           "Minimal number of lookupswitch entries for rewriting to binary " \
           "switch")                                                         \
                                                                             \
-  develop(intx, StopInterpreterAt, 0,                                       \
+  develop(int, StopInterpreterAt, 0,                                        \
```

Comment:
Some cpu ports access this from assembly, so changing the size is gong to require cpu-specific changes.  I'd rather see this and BytecodeCounter::_counter_value both changed to uint64_t.

Reply:
share/interpreter/bytecodeHistogram.cpp:int   BytecodeCounter::_counter_value = 0;

Maybe this should be inx instead.

Reply:
The only reason to use `intx` is if this should be 32-bit on 32-bit and 64-bit on 64-bit. I don't think that is the case here.

Reply:
The reason that this should be reverted is that to fix it, you have to fix all the platform code and that seems out of scope for fixing Wconversion warnings.

Reply:
Actually this counts bytecodes, so it could easily become 64 bit on 64 bit platforms.

Reply:
This is complicated.  BytecodeCounter::_counter_value is an int, and StopInterpreterAt compares to that in shared code, which is fine because it'll promote the former to intx.  The cpu-specific code fetches the former as int and the latter as intx, except in one place so fixing either is going to be work.  The one place that complains on x86 is 

src/hotspot/cpu/x86/templateInterpreterGenerator_x86.cpp:1865:12: warning: conversion from 'intx' {aka 'long int'} to 'int32_t' {aka 'int'} may change value [-Wconversion]
 1865 |            StopInterpreterAt,
      |            ^~~~~~~~~~~~~~~~~

```
  __ cmp32(ExternalAddress((address) &BytecodeCounter::_counter_value),
           StopInterpreterAt,
           rscratch1);
```
We could just static_cast StopInterpreterAt to int here.

Reply:
> Actually this counts bytecodes, so it could easily become 64 bit on 64 bit platforms.

I thought this was the count within a method, in which case the bytecode limit is the same for 32-bit and 64-bit.

Reply:
It is not.